core/src/main/java/io/questdb/griffin/engine/table/PrometheusMetricsRecordCursorFactory.java

```diff
@@ -0,0 +1,350 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.table;
+
+import io.questdb.Metrics;
+import io.questdb.cairo.AbstractRecordCursorFactory;
+import io.questdb.cairo.ColumnType;
+import io.questdb.cairo.DataUnavailableException;
+import io.questdb.cairo.GenericRecordMetadata;
+import io.questdb.cairo.TableColumnMetadata;
+import io.questdb.cairo.sql.Record;
+import io.questdb.cairo.sql.RecordCursor;
+import io.questdb.cairo.sql.RecordMetadata;
+import io.questdb.griffin.PlanSink;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.metrics.MetricsRegistry;
+import io.questdb.metrics.PrometheusFormatUtils;
+import io.questdb.metrics.Target;
+import io.questdb.std.LongList;
+import io.questdb.std.Misc;
+import io.questdb.std.str.DirectUtf8Sink;
+import io.questdb.std.str.DirectUtf8String;
+import io.questdb.std.str.Utf8Sequence;
+import io.questdb.std.str.Utf8String;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+
+import static io.questdb.metrics.MemoryTagLongGauge.MEMORY_TAG_PREFIX;
+
+public final class PrometheusMetricsRecordCursorFactory extends AbstractRecordCursorFactory {
+    public static final RecordMetadata METADATA;
+    private final PrometheusMetricsCursor prometheusMetricsCursor = new PrometheusMetricsCursor();
+
+    public PrometheusMetricsRecordCursorFactory() {
+        super(METADATA);
+    }
+
+    @Override
+    public void _close() {
+        prometheusMetricsCursor.close();
+    }
+
+    @Override
+    public RecordCursor getCursor(SqlExecutionContext sqlExecutionContext) {
+        prometheusMetricsCursor.of(sqlExecutionContext.getCairoEngine().getMetrics());
+        prometheusMetricsCursor.toTop();
+        return prometheusMetricsCursor;
+    }
+
+    @Override
+    public boolean recordCursorSupportsRandomAccess() {
+        return false;
+    }
+
+    @Override
+    public void toPlan(PlanSink sink) {
+        sink.type("prometheus_metrics");
+    }
+
+    public static class PrometheusMetricsCursor implements RecordCursor {
+        PrometheusMetricsRecord record;
+        private int pos;
+        private MetricsRegistry registry;
+        private int size;
+        private int subLimit;
+        private int subPos;
+
+        public PrometheusMetricsCursor() {
+            DirectUtf8Sink sink = new DirectUtf8Sink(255);
+            LongList values = new LongList(10);
+            DirectUtf8String value = new DirectUtf8String();
+            record = new PrometheusMetricsRecord(sink, values, value);
+        }
+
+        @Override
+        public void close() {
+            record.close();
```

Comment:
is closing the record a good idea? chances are once this factory is created it's gonna be called in regular intervals, say every few seconds. this means you are gonna inflate and deflate the native sink every few seconds. this seems like a waste, in the worst case it will cause memory fragmentation. wdyt?